feat(glean): Add 2FA 'these authentication apps' link click event, adjust DataBlock data attrs

LZoog opened the pull request:
Because:
* We want metrics for when a user clicks on this link
* Simplifies glean data attrs in DataBlock because the type is string or undefined, and if this is undefined the attribute does not render

This commit:
* Adds the glean attrs on LinkExternal
* Shares 2fa glean data 'type's for consistency

closes FXA-9578

chenba commented on packages/fxa-react/components/LinkExternal/index.tsx:
```diff
@@ -4,13 +4,18 @@
   'data-testid'?: string;
+  gleanDataAttrs?: {
+    id: string;
+    type?: GleanDataType2fa;
```
Seems strange to me that a 2fa specific type is used in a much more generic type for link props.

LZoog replied:
Sure, but wouldn't we just expand it at that time? That's what I pictured but, I can change to 'string'. (Maybe this should be an enum anyways...)

chenba replied:
> but wouldn't we just expand it at that time?

Does it make sense to have 'inline setup' always as a valid value of that type when applied to LinkExternal?  I'm arguing no and we might as well be generic.  I could see using a type argument to limit the valid values to whatever context in which the LinkExternal is used.

LZoog replied:
Eh yeah good point. Mostly I just wanted us to not typo the `type` here but I'll wait for #17459 and #17460 to land and then I'll adjust this and turn that 2fa type into an enum and update all around, will likely request re-review.

chenba commented on packages/fxa-react/lib/interfaces.ts:
```diff
@@ -0,0 +1,5 @@
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+export type GleanDataType2fa = 'setup' | 'inline setup' | 'replace';
```
I'm not afraid of spaces, but I also don't know how Glean is going to handle them.

LZoog replied:
Hmmm, well I'm certainly open to using a dash or something instead.

chenba reviewed (approved):
👍  but I think the type for the Glean "type" in LinkExternal should be generic.

LZoog commented:
Just for posterity: this was r+ but I pushed a small update addressing a comment and added that enum for 2fa Glean attr 'type's and updated the references which was discussed in Slack - since another issue depends on this one for the `LinkExternal` updates, I'll merge once CI goes green.
